src/cdk/collections/selection-model.ts

```diff
@@ -238,6 +233,21 @@ export class SelectionModel<T> {
   private _hasQueuedChanges() {
     return !!(this._deselectedToEmit.length || this._selectedToEmit.length);
   }
+
+  /** Returns a value that is comparable to inputValue by applying compareWith function, returns the same inputValue otherwise. */
+  private _getConcreteValue(inputValue: T): T {
+    let concreteValue = inputValue;
+    if (!this.compareWith) {
+      return concreteValue;
+    } else {
+      this._selection.forEach(selectedValue => {
+        if (this.compareWith?.call(undefined, inputValue, selectedValue)) {
```

Comment:
The `?.` here is going to generate extra code. You can use `!.` since we know that it'll be defined.

Reply:
Thanks, i didn't know about the ! operator in js  (corrected).